core-bundle/src/EventListener/DoctrineSchema/AdjustSearchUrlLengthListener.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\EventListener\DoctrineSchema;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Schema\Table;
+use Doctrine\ORM\Tools\Event\GenerateSchemaEventArgs;
+
+/**
+ * Adjusts the length of tl_search.url if innodb_large_prefix is not enabled (#4615).
+ *
+ * @internal
+ */
+class AdjustSearchUrlLengthListener
+{
+    private $connection;
+
+    public function __construct(Connection $connection)
+    {
+        $this->connection = $connection;
+    }
+
+    public function __invoke(GenerateSchemaEventArgs $event): void
+    {
+        if (
+            !($schema = $event->getSchema())
+            || !$schema->hasTable('tl_search')
+            || !($table = $schema->getTable('tl_search'))
+            || !$table->hasColumn('url')
+            || !($column = $table->getColumn('url'))
+            || !$table->hasIndex('url')
+        ) {
+            return;
+        }
+
+        // Get maximum index size for this table
+        $maxIndexSize = $this->getMaximumIndexSize($table);
+
+        // Reduce maximum length if collation is not "ascii_bin"
+        if ('ascii_bin' !== $column->getPlatformOption('collation')) {
```

Comment:
How can this ever be true? Our column definition contains `COLLATE ascii_bin`.

Reply:
The SQL definition of the field can be arbitrarily adjusted by extension, application customisations or directly on the database level. The listener cannot rely on the original column definition.

Reply:
Of course it can. Do you really think you can foresee every possible way to misconfigure the system?

Reply:
We check what we can check... why wouldn't we? I don't really understand why you would not want to include this check.

Reply:
The check in question also concerns case A because it also depends on the database server configuration whether or not assuming a byte length of 4 leads to an issue.

Reply:
I think these checks are relevant, because of problems with the length, some users were forced to make modifications to this field. And since we changed this to `ascii_bin` quite recently the possibility of it being set to `utf8` is quite high I think.